File: xml-performance/test-drive-xml-jaxb/src/main/java/com/steelzack/test/drive/xml/XmlBookJAXBParserManager.java
package com.steelzack.test.drive.xml;

import java.io.InputStream;

import javax.xml.bind.JAXBContext;
import javax.xml.bind.Unmarshaller;

import com.steelzack.test.drive.jaxb.books.xml.Books;

public class XmlBookJAXBParserManager extends XmlBookParserBuilder {
	private Books books;

	public XmlBookJAXBParserManager(InputStream ioStream) {
		super(ioStream);
	}

	@Override
	public String getBookTitle(int bookNumber) {
		return books.getBooks().get(bookNumber).getTitle();
	}

	@Override
	public int getNumberOfPages(int bookNumber) {
		return books.getBooks().get(bookNumber).getPages();
	}

	@Override
	public String getGenre(int bookNumber) {
		return books.getBooks().get(bookNumber).getGenre();
	}

	@Override
	public String getAuthor(int bookNumber) {
		return books.getBooks().get(bookNumber).getAuthor();
	}

	@Override
	public int getNumberOfBooks() {
		return books.getBooks().size();
	}

	@Override
	public void init() throws Exception {
		final JAXBContext context = JAXBContext.newInstance(Books.class);
		Unmarshaller u = context.createUnmarshaller();
		books = (Books) u.unmarshal(getIoStream());
	}

}

File: xml-performance/test-drive-xml-jaxb/src/main/kotlin/org/jesperancinha/test/drive/xml/XmlBookJAXBParserManager.kt
package org.jesperancinha.test.drive.xml

import jakarta.xml.bind.JAXBContext
import org.jesperancinha.test.drive.jaxb.books.xml.Books
import java.io.InputStream

class XmlBookJAXBParserManager(ioStream: InputStream) : XmlBookParserBuilder(ioStream) {
    lateinit var books: Books
    override fun getBookTitle(bookNumber: Int): String? {
        return books.books[bookNumber].title
    }

    override fun getNumberOfPages(bookNumber: Int): Int {
        return books.books[bookNumber].pages
    }

    override fun getGenre(bookNumber: Int): String? {
        return books.books[bookNumber].genre
    }

    override fun getAuthor(bookNumber: Int): String? {
        return books.books[bookNumber].author
    }

    override val numberOfBooks: Int
        get() = books.books.size

    @Throws(Exception::class)
    override fun init() {
        val context = JAXBContext.newInstance(Books::class.java)
        val u = context.createUnmarshaller()
        books = u.unmarshal(ioStream) as Books
    }
}